layout.ld

```diff
@@ -147,3 +154,6 @@ SECTIONS {
     } > FLASH
     PROVIDE_HIDDEN (__exidx_end = .);
 }
+
+ASSERT((_sstack_aligned - _s_stack_unaligned) == 0, "
+SRAM must be aligned 8 byte aligned")
```

Comment:
http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.faqs/ka4127.html

Reply:
There are two separate alignments happening in this linker script: one at the beginning of SRAM (the one this assertion checks) and one at the bottom of the stack (the stack's 8-byte alignment). I don't see why the linker script needs to care about the alignment of the start of SRAM/top of the stack as long as the bottom of the stack is 8-byte-aligned.

Reply:
(based on v3 patch series)

AFAICT, there are two conditions that we need.

1. SRAM needs to start at a 8 byte alignment. This implemented using,

```
        _stack_bottom_unaligned = .;
        . = ALIGN(8);
        _stack_bottom_aligned = .;

[...]

ASSERT((_stack_bottom_aligned - _stack_bottom_unaligned) == 0, "
SRAM must be 8 byte aligned")
```

2. `STACK_SIZE` _must_ be a multiple of 8 (otherwise SRAM region start + `STACK_SIZE` will _not_ be 8 byte aligned. This is implemented using,

```
        _stack_top_unaligned = .;
        . = ALIGN(8);
        _stack_top_aligned = .;
 
[...]

ASSERT((_stack_top_aligned - _stack_top_unaligned) == 0, "
STACK_SIZE must be 8 byte multiple")
```

Reply:
My question is: where does your condition #1 come from? As fan as I can tell, SRAM region start + STACK_SIZE + padding must be 8 byte aligned, but there isn't a need for SRAM start itself to be 8-byte aligned.

Reply:
Yes, I think you are right. We do not require #1. While having SRAM aligned to 32-byte boundary (not _8_ as previously assumed) would be nice, it is not mandatory). I'll correct it in the next revision.

I think I forgot to add "memory" clobber. I'll also add that in the next revision.